src/pj_fwd.c

```diff
@@ -81,9 +109,42 @@ PJ_COORD pj_fwd_finalize (PJ *P, PJ_COORD coo) {
     }
 
     /* Handle false eastings/northings and non-metric linear units */
-    coo.xyz.x = P->fr_meter  * (coo.xyz.x + P->x0);
-    coo.xyz.y = P->fr_meter  * (coo.xyz.y + P->y0);
-    coo.xyz.z = P->vfr_meter * (coo.xyz.z + P->z0);
+    if (P->right==PJ_IO_UNITS_CARTESIAN) {
```

Comment:
in `pj_inv` you define `INPUT_UNITS` and `OUTPUT_UNITS`. Doing that here as well will make it easier to read the code

Reply:
Yes, and I will do that. I managed to do the mental math necessary in the forward case, without having this translation/reminder, whereas when having to do the same thing `backwards and in high heels`, in the inverse case, I really needed these persistent pings